apps/hpc-ftsadmin/src/main.tsx

```
@@ -1,11 +1,13 @@
-import ReactDOM from 'react-dom';
+import ReactDOM from 'react-dom/client';
 import { BrowserRouter } from 'react-router-dom';
 
 import App from './app/app';
 
-ReactDOM.render(
+const root = ReactDOM.createRoot(
+  document.getElementById('root') as HTMLElement
+);
```

In the other comment I am saying bad things about type assertion and how it should be avoided, because TS compiler usually knows better than us. Here, that's not the case, because we (the developers) know better than Typescript. DOM element with ID `'root'` definitely exists and `document.getElementById` should not return `null`.

But still, I like to avoid type assertions, so let's throw an error if root element cannot be found.
```ts
const rootElement = document.getElementById('root');
if (!rootElement) {
  throw new Error('Failed to find the root element');
}
const root = ReactDOM.createRoot(rootElement);
```